frontend/projects/upgrade/src/app/features/dashboard/profile/components/metrics/metrics.component.ts

```diff
@@ -89,6 +90,16 @@ export class MetricsComponent implements OnInit, OnDestroy, AfterViewInit {
     return metrics;
   }
 
+  openAddMetricDialog() {
+    this.selectedMetricIndex = null;
+    const dialogRef = this.dialog.open(AddMetricsComponent, {
+      panelClass: 'add-metric-modal',
+    });
+    dialogRef.afterClosed().subscribe((result) => {
```

Comment:
Do we need this subscriber?

Reply:
currently no but if we wish do anything after closing dialog box later, we could use it like refreshing page-view maybe.